src/app/rpmostree-clientlib.cxx

```diff
@@ -1682,7 +1684,31 @@ append_docs_to_str (GString          *str,
     }
 }
 
-/* Check whether sd_unit contains pid and return the boolean in sd_unit_contains_pid */
+/* Check whether sd_unit's `ActiveState` is in `states` and return the boolean
+ * in `sd_unit_is_active`. */
+static gboolean
+check_sd_unit_state (char            *sd_unit,
+                     const char     **states,
+                     gboolean        *states_contains_state,
```

Comment:
Minor/optional: hmm interesting API. :)  It's neat that it's super generic like this, though for our purposes here IMO it'd be cleaner if we just folded in the array in this function and just returned a "is_active" boolean. Cool as is too though!

Reply:
I thought about this a bit, mainly because I was wondering what I should name the helper function if we include the "failed" state in the list.
But I guess it's unlikely that the helper will be used anywhere else, so perhaps just a `check_sd_unit_is_active` that does everything with a note about the additional "failed" case would be fine.